coreax/solvers/recombination.py

```diff
@@ -259,7 +279,7 @@ def _eliminate(state: _EliminationState) -> _EliminationState:
 
 def _push_forward(
     nodes: Shaped[Array, "n d"],
-    test_functions: Union[Callable[[Omega], Real[Array, " m-1"]], None],
+    test_functions: Optional[Callable[[Omega], Real[Array, " m-1"]]],
```

Comment:
Seeing "Type variable "Omega" has no meaning in this context", looks like this could be causing an issue at line 302 as well

Reply:
If this is always going to be an array in practice it is probably be better to type it as that and leave the $\Omega$ as just docstring notation. What do you thinkk?

Reply:
Ignoring any precise technicalities, `Omega` can be any space, providing it is the domain of the test-functions. However, in practice, due to the usage of `jax.vmap` over the `test_functions`, our `nodes` must always be some `jax.Array` of shape `[n, ...]`. We could handle the more general case, by allowing the `test_functions` to be specified as "pre-batched", but this would add considerable complexity for what in practice is probably a niche use case (if user requirements change, a new issue can be raised in the future to provide such handling).

My proposal is to replace `Omega` with the more precise `_ElementOfOmega` type variable and bound this by `jax.Array` (the type variable shows that the nodes must be an element of the test-function domain), then in `RecombinationSolver.test_functions` we simply use the `jax.Array` type hint as you suggested. Does this make sense/are you happy with this?